softwareComponents/simplesimClientLib/src/changecolor.cpp

```diff
@@ -0,0 +1,97 @@
+#include "changecolor.hpp"
+#include "ui_changecolor.h"
+
+#include <iostream>
+
+using rofi::simplesim::ChangeColor;
+
+ChangeColor::ChangeColor( QWidget* parent, size_t size ) :
+    ui( new Ui::ChangeColor ),
+    parent( parent ),
+    to_color( size, false )
+{
+    //QWidget( parent );
+    ui->setupUi( this );
+    ui->plainTextEdit->hide();
+
+    connect( ui->buttonBox, SIGNAL( rejected() ), this, SLOT( hide() ) );
+    connect( ui->buttonBox, SIGNAL( accepted() ), this, SLOT( accept() ) );
+    connect( ui->toolButton, SIGNAL( clicked() ), this, SLOT( showHelp() ) );
+}
+
+ChangeColor::~ChangeColor(){
+    delete ui;
+}
+
+void ChangeColor::accept(){
+    hide();
+    try {
+        parseRange();
+    } catch ( std::exception& e ){
+        std::cout << "couldn't parse input\n";
+        QString empty;
+        ui->textEdit->setPlainText( empty );
+    }
+
+    int color = ui->listWidget->currentRow();
+    if( color == -1 ){
+        std::cout << "no color selected\n";
+    }
+    emit pickedColor( color );
+}
+
+void ChangeColor::showHelp(){
+    if( showingHelp ){
+        ui->plainTextEdit->hide();
+    } else {
+        ui->plainTextEdit->show();
+    }
+    showingHelp = !showingHelp;
+}
+void ChangeColor::parseRange(){
+    QString source = ui->textEdit->toPlainText();
+    int i = 0;
+    int number = -1;
+
+    auto parseNum = [&](){
+        std::string buffer;
+        if( !source[ i ].isDigit() ){
+            throw std::exception();
```

Comment:
Throwing `std::exception` is bad practice. Use at least `std::logic_error` or `std::runtime_error`.